README
Gnome2::Canvas version 1.005
============================

NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE

This module has been deprecated by the Gtk-Perl project.  This means that the
module will no longer be updated with security patches, bug fixes, or when
changes are made in the Perl ABI.  The Git repo for this module has been
archived (made read-only), it will no longer possible to submit new commits to
it.  You are more than welcome to ask about this module on the Gtk-Perl
mailing list, but our priorities going forward will be maintaining Gtk-Perl
modules that are supported and maintained upstream; this module is neither.

Since this module is licensed under the LGPL v2.1, you may also fork this
module, if you wish, but you will need to use a different name for it on CPAN,
and the Gtk-Perl team requests that you use your own resources (mailing list,
Git repos, bug trackers, etc.) to maintain your fork going forward.

* Perl URL: https://gitlab.gnome.org/GNOME/perl-gnome2-canvas
* Upstream URL: https://gitlab.gnome.org/Archive/libgnomecanvas
* Last upstream version: 2.30.3
* Last upstream release date: 2011-01-31
* Migration path for this module: Cairo
* Migration module URL: https://metacpan.org/pod/Cairo

NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE NOTICE


The Gnome2::Canvas module allows a perl developer to use the GnomeCanvas
widget with Gtk2-Perl.  Find out more about Gnome+ at http://www.gnome.org.

Like the Gtk2 module on which it depends, Gnome2::Canvas follows the C API
of libgnomecanvas-2.0 as closely as possible while still being perlish.
Thus, the C API reference remains the canonical documentation.

To discuss gtk2-perl, ask questions and flame/praise the authors,
join gtk-perl-list@gnome.org at lists.gnome.org.

Also have a look at the gtk2-perl website and sourceforge project page,
http://gtk2-perl.sourceforge.net


INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install


To avoid installing to a system directory, since this is a beta release,
you can change the installation prefix at Makefile.PL time with

   perl Makefile.PL PREFIX=/some/other/place

This will install the module to the subdirectory lib/perl5 under the given
prefix.  If this is not already in perl's include path, you'll need to tell
perl how to get to this library directory so you can use it; there are three
ways:

  in your environment (the easiest):
    # assuming a bourne-style shell
    PERL5LIB=/some/other/place/lib/perl5/site_perl
    export PERL5LIB

  on the perl command line:
    perl -I /some/other/place/lib/perl5/site_perl yourscript

  in the code of your perl script:
    use lib '/some/other/place/lib/perl5/site_perl';
  

DEPENDENCIES

This module requires these other modules and libraries:

  perl >= 5.8.0
  Glib >= 1.040 (perl module and requisite C libraries)
  Gtk2 >= 1.040 (perl module and requisite C libraries)
  libgnomecanvas > 2.x (C library)

In order to build it from source, you'll also need

  ExtUtils::Depends >= 0.2
  ExtUtils::PkgConfig >= 1.03


COPYRIGHT AND LICENSE

Copyright (C) 2003-2004 by the gtk2-perl team (see the file AUTHORS for a
complete list)

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Library General Public
License as published by the Free Software Foundation; either
version 2 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
Library General Public License for more details.

You should have received a copy of the GNU Library General Public
License along with this library; if not, see
<https://www.gnu.org/licenses/>.


NEWS
Overview of Changes in 1.005
============================

* Deprecation release 1.005, this module is now no longer maintained.
  See the README file for more deprecation information.

Overview of Changes in 1.004
============================

* Makefile.PL: added 'NO_META => 1'; Calling 'make dist' is clobbering the
  META.yml file that's already included in the distribution, which is wrong
  and breaks things

Overview of Changes in 1.003
============================

* META.yml: tweak file to fix problems with using 'cpanm' to install module;
  Moved 'Gtk2' from 'requires:' to 'configure_requires:'; Set the 'version:'
  string to the current module version, instead of 'unknown'; both of these
  changes fix RT#127878
* Updated FSF address in all files that have it; this closes RT#111436

Overview of Changes in 1.002
============================

* Fix a bug in Gnome2::Canvas::polygon_to_point().  [muppet]
* Fix unusably broken bindings of Gnome2::Canvas::w2c_affine(),
  Gnome2::Canvas::Item::i2w_affine(), and Gnome2::Canvas::Item::i2c_affine(),
  with regression tests.  [muppet]
* Portability fixes for Mac OS X.  [muppet]
* Minor documentation enhancements.  [muppet]

Overview of Changes in 1.001
============================

* Fix the broken example in the documentation synopsis.  [muppet]
* Skip the open_parts and closed_parts tests to pass when working with
  libgnomecanvas 2.0.x.  [Torsten]
* Hush unused variable warning.  [muppet]

Overview of Changes in 1.0
==========================

* Require stable Glib/Gtk2 1.040
* Add import version checking.
* Spit and polish.

Overview of Changes in 0.93
===========================

* Check that we're using at least ExtUtils::Depends 0.2.
* Require ExtUtils::PkgConfig 1.03.
* Add Gnome2::Canvas::GET_VERSION and Gnome2::CHECK_VERSION. [Emmaneule Bassi]
* Pod improvements.

Overview of Changes in Gnome2::Canvas 0.92
==========================================

* Updated Glib requirement to 1.035.
* Made use of Glib 1.035's MakeHelper features for documentation generation.

Overview of Changes in Gnome2::Canvas 0.91
==========================================

* GnomeCancasPathDef, handling re-worked.
* apidoc POD updates
* some new demos added, others updated


Makefile.PL
#
# $Id$
#

use strict;
use 5.008;
use ExtUtils::MakeMaker;
use Cwd;
use File::Spec;

# minimum required version of dependencies we need to build
our %build_reqs = (
	'perl-ExtUtils-Depends'   => '0.200',
	'perl-ExtUtils-PkgConfig' => '1.03',
	'perl-Glib'               => '1.040',
	'perl-Gtk2'               => '1.040',
	'GnomeCanvas'             => '2.0.0',
);

# Writing a fake Makefile ensures that CPAN will pick up the correct
# dependencies and install them.
unless (eval "use ExtUtils::Depends '$build_reqs{'perl-ExtUtils-Depends'}';"
           . "use ExtUtils::PkgConfig;"
           # just seeing if Glib is available isn't enough, make sure
           # it's recent enough, too 
           . "use Glib '$build_reqs{'perl-Glib'}';"
	   . "use Glib::MakeHelper;"
           . "use Gtk2 '$build_reqs{'perl-Gtk2'}';"
           . "use Gtk2::CodeGen;"
           . "1") {
   warn "$@\n";
   WriteMakefile(
         PREREQ_FATAL => 1,
         NO_META      => 1,
         PREREQ_PM    => {
            Glib::                => $build_reqs{'perl-Glib'},
            Gtk2::                => $build_reqs{'perl-Gtk2'},
            ExtUtils::Depends::   => $build_reqs{'perl-ExtUtils-Depends'},
            ExtUtils::PkgConfig:: => $build_reqs{'perl-ExtUtils-PkgConfig'},
         },
   );
   exit 1; # not reached
}

my %pkgcfg = ExtUtils::PkgConfig->find ('libgnomecanvas-2.0 >= '
				       . $build_reqs{GnomeCanvas});

mkdir 'build', 0777;

#
# autogeneration
#
Gtk2::CodeGen->parse_maps ('gnomecanvasperl');
Gtk2::CodeGen->write_boot (ignore => '^Gnome2::Canvas$');
ExtUtils::PkgConfig->write_version_macros ("build/gnomecanvasperl-version.h",
                                           'libgnomecanvas-2.0' => 'GNOME_CANVAS');

our @xs_files = <xs/*.xs>;
our %pm_files = ('Canvas.pm' => '$(INST_LIBDIR)/Canvas.pm',);
our %pod_files = Glib::MakeHelper->do_pod_files (@xs_files);
our @typemaps = qw(canvas.typemap build/gnomecanvasperl.typemap);
our @headers = qw(gnomecanvasperl.h
                  build/gnomecanvasperl-autogen.h
                  build/gnomecanvasperl-version.h);


# now we're ready to start creating the makefile.
# we need to use ExtUtils::Depends to get relevant information out of
# the Glib extension, and to save config information for other modules which
# will chain from this one.

my $gnomecanvas = ExtUtils::Depends->new ('Gnome2::Canvas', 'Gtk2');
$gnomecanvas->set_inc ($pkgcfg{cflags});
$gnomecanvas->set_libs ($pkgcfg{libs});
$gnomecanvas->add_xs (@xs_files);
$gnomecanvas->add_pm (%pm_files);
my $cwd = cwd();
$gnomecanvas->add_typemaps (map {File::Spec->catfile($cwd,$_)} @typemaps);

$gnomecanvas->install (@headers);
$gnomecanvas->save_config ('build/IFiles.pm');

WriteMakefile(
    NAME            => 'Gnome2::Canvas',
    VERSION_FROM    => 'Canvas.pm', # finds $VERSION
    ABSTRACT        => '(DEPRECATED) Perl bindings for the 2.x series of the Gnome Canvas widget',
    XSPROTOARG      => '-noprototypes',
    MAN3PODS        => \%pod_files,
    NO_META         => 1,
    $gnomecanvas->get_makefile_vars,
);


sub MY::postamble {
	return Glib::MakeHelper->postamble_clean ()
	     . Glib::MakeHelper->postamble_docs_full (
	         DEPENDS => $gnomecanvas,
		 COPYRIGHT => "Copyright (C) 2003-2004 by the Gtk2-Perl Team.\n\nThis software is licensed under the LGPL; see L<Gnome2::Canvas> for a full notice.",
	       )
	     . Glib::MakeHelper->postamble_rpms (
		'GNOME_CANVAS' => $build_reqs{'GnomeCanvas'},
		'PERL_EXTUTILS_DEPENDS' => 
			$build_reqs{'perl-ExtUtils-Depends'},
		'PERL_EXTUTILS_PKGCONFIG' => 
			$build_reqs{'perl-ExtUtils-PkgConfig'},
		'PERL_GLIB' => $build_reqs{'perl-Glib'},
		'PERL_GTK' => $build_reqs{'perl-Gtk2'},
	       );
}
